src/masoniteorm/relationships/HasManyThrough.py

```diff
@@ -157,47 +263,3 @@ def get_with_count_query(self, builder, callback):
         )
 
         return return_query
-
-    def attach(self, current_model, related_record):
-        raise NotImplementedError(
-            "HasOneThrough relationship does not implement the attach method"
-        )
-
-    def attach_related(self, current_model, related_record):
-        raise NotImplementedError(
-            "HasOneThrough relationship does not implement the attach_related method"
-        )
-
-    def query_has(self, current_query_builder, method="where_exists"):
-        related_builder = self.get_builder()
-
-        getattr(current_query_builder, method)(
-            self.distant_builder.where_column(
-                f"{current_query_builder.get_table_name()}.{self.local_owner_key}",
-                f"{self.intermediary_builder.get_table_name()}.{self.local_key}",
-            ).join(
-                f"{self.intermediary_builder.get_table_name()}",
-                f"{self.intermediary_builder.get_table_name()}.{self.foreign_key}",
-                "=",
-                f"{self.distant_builder.get_table_name()}.{self.other_owner_key}",
-            )
-        )
-
-        return related_builder
-
-    def query_where_exists(
-        self, current_query_builder, callback, method="where_exists"
-    ):
-        query = self.distant_builder
-
-        getattr(current_query_builder, method)(
-            query.join(
-                f"{self.intermediary_builder.get_table_name()}",
-                f"{self.intermediary_builder.get_table_name()}.{self.foreign_key}",
-                "=",
-                f"{query.get_table_name()}.{self.other_owner_key}",
-            ).where_column(
-                f"{current_query_builder.get_table_name()}.{self.local_owner_key}",
-                f"{self.intermediary_builder.get_table_name()}.{self.local_key}",
-            )
-        ).when(callback, lambda q: (callback(q)))
```

Comment:
why are all these deleted?

Reply:
They are not deleted.
These lines are now at 221 - 225 respectively
```python
                f"{intermediate_table}.{self.local_key}",
                f"{current_builder.get_table_name()}.{self.local_owner_key}",
            )
            .when(callback, lambda q: (callback(q)))
        )

Lines  200 & 201 had readability changes.
Line 203 moved the `.when()` from the current query builder to the distant builder as thats where the criteria should be